ui/src/userInput/AzExtUserInput.ts

```diff
@@ -24,7 +24,7 @@ export class AzExtUserInput implements types.IAzureUserInput {
         return this._onDidFinishPromptEmitter.event;
     }
 
-    public async showQuickPick<TPick extends QuickPickItem>(picks: TPick[] | Promise<TPick[]>, options: types.IAzureQuickPickOptions): Promise<TPick | TPick[]> {
+    public async showQuickPick<TPick extends types.IAzureQuickPickItem<unknown>>(picks: TPick[] | Promise<TPick[]>, options: types.IAzureQuickPickOptions): Promise<TPick | TPick[]> {
```

Comment:
Could you explain the `QuickPickItem` ->  `types.IAzureQuickPickItem<unknown>` changes?

Reply:
Inside `bumpRecentlyUsedPick` I needed to access `suppressPersistence`, but it only exists on our interface `IAzureQuickPickItem`, not VS Code's `QuickPickItem`. I could've used a cast, but I'm not a fan of casting. So I changed the type in that method and kept changing it up the stack until all the build errors went away

Also I just realized I could get rid of some old casts due to my changes so I just cleaned that up.